pkg/collector/reconcile/configmap.go

```diff
@@ -120,7 +120,7 @@ func expectedConfigMaps(ctx context.Context, params Params, expected []corev1.Co
 			updated.ObjectMeta.Labels[k] = v
 		}
 
-		patch := client.MergeFrom(&params.Instance)
+		patch := client.MergeFrom(existing)
```

Comment:
This was indeed a bug! Looks like the same problem exists for the Service and ServiceAccount.

Reply:
Yes it does. I ll raise seperate PR for it.